radicle-cli/src/commands/remote/list.rs

```diff
@@ -0,0 +1,44 @@
+use std::str::FromStr;
+
+use radicle_term::{Element, Table};
+
+use crate::git;
+use crate::terminal as term;
+
+#[inline]
+fn format_direction(d: &git::Direction) -> String {
+    match d {
+        git::Direction::Fetch => "fetch".to_owned(),
+        git::Direction::Push => "push".to_owned(),
+    }
+}
+
+pub fn run(repo: &git::Repository) -> anyhow::Result<()> {
+    let mut table = Table::default();
+    let remotes = git::rad_remotes(repo)?;
+    for r in remotes {
+        for spec in r.refspecs() {
+            let dir = spec.direction();
+            let url = r.url().ok_or(anyhow::anyhow!("url not present"))?;
+            let name = r
+                .name()
+                .ok_or(anyhow::anyhow!("remote name is not present"))?;
+            table.push([
+                term::format::badge_positive(format_direction(&dir)),
+                term::format::highlight(name.to_owned()),
+                term::format::highlight(url_to_nid(url)?),
+            ]);
+        }
+    }
+    table.print();
+    Ok(())
+}
+
+fn url_to_nid(url: &str) -> anyhow::Result<String> {
+    let url = radicle::git::Url::from_str(url)?;
+    if let Some(nid) = url.namespace {
+        return Ok(nid.to_string());
+    }
+    // FIXME: when the namespace is null?
+    Ok("".to_owned())
+}
```

Comment:
Looks like something is wrong here, why are we returning an empty string? And why is a function called `url_to_nid` not returning a `NodeId`?